From 7061d75fc7a623daea7c40b27532a21563de6f2b Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Tue, 25 Dec 2018 10:55:42 +0800
Subject: [PATCH 426/706] tsn: felix: DEI map DP value support

Mapping CFI of Vlan to DP value, switch ports can recognize green and
yellow frames from outside. It can be used by EIR of PSFP policer.

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
(cherry picked from commit ac9370472d2826afde837886f9851fd9808b288a)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/tsn_switch.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/tsn_switch.c b/drivers/net/ethernet/mscc/tsn_switch.c
index 850f63e33d5b..c636066ab4be 100644
--- a/drivers/net/ethernet/mscc/tsn_switch.c
+++ b/drivers/net/ethernet/mscc/tsn_switch.c
@@ -1044,9 +1044,11 @@ int switch_pcp_map_set(struct net_device *ndev, bool enable)
 		       ANA_PORT_QOS_CFG,
 		       port->chip_port);
 
-	for (i = 0; i < NUM_MSCC_QOS_PRIO; i++) {
+	for (i = 0; i < NUM_MSCC_QOS_PRIO * 2; i++) {
 		ocelot_rmw_ix(ocelot,
+			      (ANA_PORT_PCP_DEI_MAP_DP_PCP_DEI_VAL & i) |
 			      ANA_PORT_PCP_DEI_MAP_QOS_PCP_DEI_VAL(i),
+			      ANA_PORT_PCP_DEI_MAP_DP_PCP_DEI_VAL |
 			      ANA_PORT_PCP_DEI_MAP_QOS_PCP_DEI_VAL_M,
 			      ANA_PORT_PCP_DEI_MAP,
 			      port->chip_port, i);
-- 
2.17.1

